### PR TITLE
Issue #2943: Add check to user page callback to prevent redirects.

### PR DESCRIPTION
**Pull request by quicksketch**
Fixes https://github.com/backdrop/backdrop-issues/issues/2943.

### CODE REVIEW

File: core/modules/user/user.module [hunk 1]
```diff
@@ -3024,9 +3024,10 @@ function user_modules_uninstalled($modules) {
  * proper destination after a user has been properly logged in.
  */
 function user_login_destination() {
+  global $user;
   $destination = backdrop_get_destination();
   if ($destination['destination'] == 'user/login') {
-    $destination['destination'] = 'user';
+    $destination['destination'] = 'user/' . $user->uid;
```
Comment (opi) [inline core/modules/user/user.module:3030]: Maybe check for `$user->uid` first ?

File: core/modules/user/user.pages.inc [hunk 2]
```diff
@@ -557,6 +557,14 @@ function user_cancel_confirm($account, $timestamp = 0, $hashed_pass = '') {
  */
 function user_page() {
   global $user;
+
+  // To help users upgrading from previous versions of Backdrop where this
```
Comment (opi) [inline core/modules/user/user.pages.inc]: - s/previous/older than 1.9.0/

- Maybe mention the https://github.com/backdrop/backdrop-issues/issues/260 issue number

**Review by opi — COMMENTED** [review]
Untested yet, but LGTM

File: core/modules/user/user.module (hunk 1 above)
Comment (jenlampton) [reply core/modules/user/user.module:3030]: >Third-party authentication modules may use this function to determine the proper destination after a user has been properly logged in. 

After you've been "properly logged in", it's safe to assume that `$user->uid` already exists.

**Comment by jenlampton** [comment]
I've tested this PR on two of my sites during the update to 1.9.0 and in both cases it solved the problem I faced last time. I think this is a nice addition for user/developer experience, and is well worth having a `menu_rebuid` happen in a strange place.

File: core/modules/user/user.module (hunk 1 above)
Comment (opi) [reply core/modules/user/user.module:3030]: Fair enough

**Comment by quicksketch** [comment]
Merged into 1.x and 1.9.x.

**Pull request merged**